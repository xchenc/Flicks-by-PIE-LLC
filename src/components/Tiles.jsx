import PropTypes from 'prop-types';
import SwiperCore, { Navigation } from 'swiper';
import { Swiper, SwiperSlide } from 'swiper/react';

import styles from 'styles/components/Tiles.module.scss';

SwiperCore.use([Navigation]);

const Tiles = ({ slidesPerView }) => (
  <Swiper
    spaceBetween={20}
    slidesPerView={1}
    slidesPerGroup={1}
    navigation
    breakpoints={{
      768: {
        slidesPerView: slidesPerView,
        slidesPerGroup: slidesPerView,
      },
    }}
    className={styles.tiles}
  >
    <SwiperSlide>
      <a href='https://nextjs.org/docs' className={styles.card}>
        <h3>Documentation &rarr;</h3>
        <p>Find in-depth information about Next.js features and API.</p>
      </a>
    </SwiperSlide>
    <SwiperSlide>
      <a href='https://nextjs.org/learn' className={styles.card}>
        <h3>Learn &rarr;</h3>
        <p>Learn about Next.js in an interactive course with quizzes!</p>
      </a>
    </SwiperSlide>
    <SwiperSlide>
      <a href='https://github.com/vercel/next.js/tree/master/examples' className={styles.card}>
        <h3>Examples &rarr;</h3>
        <p>Discover and deploy boilerplate example Next.js projects.</p>
      </a>
    </SwiperSlide>
    <SwiperSlide>
      <a
        href='https://vercel.com/new?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app'
        className={styles.card}
      >
        <h3>Deploy &rarr;</h3>
        <p>Instantly deploy your Next.js site to a public URL with Vercel.</p>
      </a>
    </SwiperSlide>
  </Swiper>
);

Tiles.propTypes = {
  slidesPerView: PropTypes.number,
};

export default Tiles;
