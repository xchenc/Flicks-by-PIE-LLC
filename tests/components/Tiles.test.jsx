import { render, fireEvent } from '@testing-library/react';
import Tiles from 'components/Tiles';

describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Tiles slidesPerView={2} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
