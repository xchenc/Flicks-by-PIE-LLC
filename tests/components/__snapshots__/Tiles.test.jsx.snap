// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="swiper-container swiper-container-initialized swiper-container-horizontal tiles"
  >
    <div
      class="swiper-button-prev"
    />
    <div
      class="swiper-button-next"
    />
    <div
      class="swiper-wrapper"
      style="transition-duration: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
    >
      <div
        class="swiper-slide"
      >
        <a
          class="card"
          href="https://nextjs.org/docs"
        >
          <h3>
            Documentation →
          </h3>
          <p>
            Find in-depth information about Next.js features and API.
          </p>
        </a>
      </div>
      <div
        class="swiper-slide"
      >
        <a
          class="card"
          href="https://nextjs.org/learn"
        >
          <h3>
            Learn →
          </h3>
          <p>
            Learn about Next.js in an interactive course with quizzes!
          </p>
        </a>
      </div>
      <div
        class="swiper-slide"
      >
        <a
          class="card"
          href="https://github.com/vercel/next.js/tree/master/examples"
        >
          <h3>
            Examples →
          </h3>
          <p>
            Discover and deploy boilerplate example Next.js projects.
          </p>
        </a>
      </div>
      <div
        class="swiper-slide"
      >
        <a
          class="card"
          href="https://vercel.com/new?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
        >
          <h3>
            Deploy →
          </h3>
          <p>
            Instantly deploy your Next.js site to a public URL with Vercel.
          </p>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
